ledger/src/blockstore.rs

```diff
@@ -3317,6 +3317,55 @@ impl Blockstore {
         );
         Ok(())
     }
+
+    /// Mark a root `slot` as connected, traverse `slot`'s children and update
+    /// the children's connected status if appropriate.
+    ///
+    /// A ledger with a full path of blocks from genesis to the latest root will
+    /// have all of the rooted blocks marked as connected such that new blocks
+    /// could also be connected. However, starting from some root (such as from
+    /// a snapshot) is a valid way to join a cluster. For this case, mark this
+    /// root as connected such that the node that joined midway through can
+    /// have their slots considered connected.
+    pub fn set_and_chain_connected_on_root(&self, slot: Slot) -> Result<()> {
+        info!(
+            "Marking slot {} and any full children slots as connected",
+            slot
+        );
+
+        let mut meta = self
+            .meta(slot)?
+            .unwrap_or_else(|| SlotMeta::new(slot, None));
+        // If the slot was already connected, there is nothing to do as this slot's
+        // children are also assumed to be appropriately connected
+        let already_connected = meta.is_connected();
+        if already_connected {
+            return Ok(());
+        }
+
+        meta.set_connected(true);
+        let mut write_batch = self.db.batch()?;
+        write_batch.put::<cf::SlotMeta>(meta.slot, &meta)?;
+
+        let mut next_slots = VecDeque::from(meta.next_slots);
+        while !next_slots.is_empty() {
+            let current_slot = next_slots.pop_front().unwrap();
+            let mut current_meta = self.meta(current_slot)?.unwrap_or_else(|| {
```

Comment:
nit: `next_meta`

Reply:
hmm, after i read the code more closely, i think just `slot`/`meta` is also okay. (i.e. without `current_`)